### PR TITLE
[CELEBORN-1746] Reduce the size of aws dependencies

### PR DESCRIPTION
**Pull request by zhaohehuhu**
### What changes were proposed in this pull request?
Due to the large size of the AWS cloud vendor's client JARs, this PR aims to keep AWS s3 module only to reduce the AWS dependency size from over 296MB to around 2.3MB

### How was this patch tested?

Screenshot 2024-11-25 at 16 17 52
tested on lab shown as above picture

### CODE REVIEW

**Comment by zhaohehuhu** [comment]
@FMX  plz help review

**Comment by FMX** [comment]
Would you like some help creating a new issue to track this PR?

**Comment by FMX** [comment]
Look like there are dependency check failures, plz fix them.

File: dev/dependencies.sh [hunk 1]
```diff
@@ -32,13 +32,14 @@ SBT_ENABLED="false"
 REPLACE="false"
 CHECK="false"
 MODULE=""
+PROFILE="aws"
```
Comment (FMX) [inline dev/dependencies.sh:35]: In default, we won't enable aws profile.

Comment (zhaohehuhu) [reply dev/dependencies.sh]: > In default, we won't enable aws profile.

Fixed. Thanks

**Comment by FMX** [comment]
This PR reduced the binary package with AWS from 481M to 194M. I'll merge it after the tests are done.

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**